common/src/test/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsEventTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.revenuecat.purchases.common.diagnostics
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.common.DateProvider
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+import java.util.Date
+
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class DiagnosticsEventTest {
+
+    private val testDate = Date(1675954145L) // Thursday, February 9, 2023 2:49:05 PM GMT
+
+    private lateinit var testDateProvider: DateProvider
+
+    @Before
+    fun setup() {
+        testDateProvider = object : DateProvider {
+            override val now: Date
+                get() = testDate
+        }
+    }
+
+    @Test
+    fun `toString transforms log event to correct JSON`() {
+        val event = DiagnosticsEvent.Log(
+            name = DiagnosticsLogEventName.ENDPOINT_HIT,
+            properties = mapOf("test-key-1" to "test-value-1", "test-key-2" to 123, "test-key-3" to true),
+            dateProvider = testDateProvider
+        )
+        val eventAsString = event.toString()
+        val expectedString = "{" +
+            "\"version\":1," +
+            "\"type\":\"log\"," +
+            "\"name\":\"endpoint_hit\"," +
+            "\"properties\":{\"test-key-1\":\"test-value-1\",\"test-key-2\":123,\"test-key-3\":true}," +
+            "\"timestamp\":1675954145" +
+            "}"
```

Comment:
@thecadams lmk if you think a body like this would be ok for sending logs to the backend. I can change it if needed though.

Reply:
Looks good.

Quick note - the real endpoint will not accept those `test-key-{1,2,3}` keys; have a look at https://github.com/RevenueCat/khepri/blob/main/khepri/controllers/diagnostics.py#L10 to see what it will accept.

(it was done in line with https://revenuecat.slack.com/archives/C04LXLZL8AX/p1676563654177199)

Reply:
Yup, this is only for unit tests. I separated the events properties a bit better in https://github.com/RevenueCat/purchases-android/pull/803 so they are the ones we expect